File: src/sauna.py
n = int(input())
ma, mb = -1, 200001
for _ in range(n):
    a, b = [int(d) for d in input().split()]
    if a > ma:
        ma = a
    if b < mb:
        mb = b
    if ma > mb:
        break
if ma <= mb:
    print(mb - ma + 1, ma)
else:
    print("bad news")

File: src/sauna.lua
n = io.read("*n")
ma, mb = -1, 200001
for _ = 1, n do
    a, b = io.read("*n", "*n")
    if a > ma then
        ma = a
    end
    if b < mb then
        mb = b
    end
    if ma > mb then
        break
    end
end
if ma <= mb then
    print(mb - ma + 1, ma)
else
    print('bad news')
end
